pytorch_lightning/trainer/training_io.py

```diff
@@ -263,12 +266,11 @@ def save_checkpoint(self, filepath, weights_only: bool = False):
             # do the actual save
             try:
                 self._atomic_save(checkpoint, filepath)
-            except AttributeError as e:
-                if 'hparams' in checkpoint:
-                    del checkpoint['hparams']
-                rank_zero_warn('warning, `hparams` dropped from checkpoint.'
-                               f' An attribute is not picklable {e}')
-
+            except AttributeError as err:
+                if 'module_arguments' in checkpoint:
```

Comment:
we should not hardcode these strings and instead define them in a constant at e.g. top of the file in lightning.py
the same name is used in multiple locations